pkg/monitoring/domainstats/downwardmetrics/scraper.go

```diff
@@ -30,25 +31,49 @@ type StaticHostMetrics struct {
 }
 
 type Scraper struct {
-	isolation            isolation.PodIsolationDetector
-	staticHostInfo       *StaticHostMetrics
-	hostMetricsCollector *hostMetricsCollector
+	isolation isolation.PodIsolationDetector
+	reporter  *DownwardMetricsReporter
 }
 
 func (s *Scraper) Scrape(socketFile string, vmi *k6sv1.VirtualMachineInstance) {
 	if !vmi.IsRunning() || !downwardmetrics.HasDownwardMetricDisk(vmi) {
 		return
 	}
 
+	metrics, err := s.reporter.Report(socketFile)
+	if err != nil {
+		log.Log.Reason(err).Infof("failed to collect the metrics")
+		return
+	}
+
+	res, err := s.isolation.Detect(vmi)
+	if err != nil {
+		log.Log.Reason(err).Infof("failed to detect root directory of the vmi pod")
+		return
+	}
+
+	metricsUpdater := vhostmd.NewMetricsIODisk(downwardmetrics.FormatDownwardMetricPath(res.Pid()))
+	err = metricsUpdater.Write(metrics)
+	if err != nil {
+		log.Log.Reason(err).Infof("failed to write metrics to disk")
+		return
+	}
+}
+
+type DownwardMetricsReporter struct {
+	staticHostInfo       *StaticHostMetrics
+	hostMetricsCollector *hostMetricsCollector
+}
+
+func (r *DownwardMetricsReporter) Report(socketFile string) (*api.Metrics, error) {
```

Comment:
it's need to be public since we call it from sever. Also, just for now I prefer to keep the name as is. In a follow-up, we need to deprecated/remove the code that delivers the metrics using a disk, so we will need to refactor this code anyway.